Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp

```diff
@@ -2581,8 +2582,11 @@ void webkitWebViewSetIcon(WebKitWebView* webView, const LinkIcon& icon, API::Dat
 }
 #endif
 
-RefPtr<WebPageProxy> webkitWebViewCreateNewPage(WebKitWebView* webView, const WindowFeatures& windowFeatures, WebKitNavigationAction* navigationAction)
+RefPtr<WebPageProxy> webkitWebViewCreateNewPage(WebKitWebView* webView, Ref<API::PageConfiguration>&& configuration, WindowFeatures&& windowFeatures, WebKitNavigationAction* navigationAction)
 {
+    ASSERT(!webView->priv->configurationForNextRelatedView);
+    SetForScope configurationForNextRelatedViewGuard(webView->priv->configurationForNextRelatedView, WTFMove(configuration));
+
```

Comment:
The received configuration is a copy of the related page configuration only changing the opener process when site isolation is enabled. So, in this function we can add another check to make sure the opener process matches too.

Reply:
I don't understand. The opener process is the only thing that's different, so we should check that it's not different?

Reply:
I misunderstood this. The opener process is the web process that calls window.open(), triggering creation of the new web process. The opener process of the new web view should match that specified by the API::PageConfiguration, or we know the application using WebKit didn't create the web view properly.